tik/lib/CartographerKernel.cpp

```diff
@@ -377,8 +422,22 @@ namespace TraceAtlas::tik
                         auto p = i->getParent();
                         if (blocks.find(p) == blocks.end())
                         {
+                            int64_t ID = IDState::Uninitialized;
                             //the use is external therefore it should be a kernel export
-                            KernelExports.push_back(GetValueID(inst));
+                            if (GetValueID(p) >= IDState::Valid)
```

Comment:
Similarly, this is inelegant. Nonnegative instead

Reply:
So I made this condition to keep the ID comparisons within the domain of the IDState enum. This is doing what you are asking for, except it is now part of the same state machine